llvm/lib/Target/AIE/AIEInterBlockScheduling.cpp

```diff
@@ -79,10 +86,19 @@ void emitBundlesInScoreboardDelta(
   }
 }
 
-MachineBasicBlock *getSinglePredecessor(const MachineBasicBlock &MBB) {
-  assert(MBB.pred_size() == 1 && "MBB contains more than 1 predecessor");
-  MachineBasicBlock *SinglePredMBB = *MBB.predecessors().begin();
-  return SinglePredMBB;
+MachineBasicBlock *getLoopPredecessor(const MachineBasicBlock &MBB) {
```

Comment:
nit: good candidate for LoopUtils.

Reply:
I'm hesitant, since it only applies to epilogues that have been selected by the classification of loop-aware candidates here.